src/vs/workbench/api/node/extHostLanguageFeatures.ts

```diff
@@ -214,6 +214,33 @@ class ReferenceAdapter {
 	}
 }
 
+class ExternalReferenceAdapter {
+
+	constructor(private _provider: vscode.ExternalReferenceProvider) {
+	}
+
+	provideExternalReferences(query: modes.ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: modes.IReferenceInformation[]) => void): TPromise<modes.IReferenceInformation[]> {
+		const convertReferences = (references: vscode.ReferenceInformation[]) => {
+			if (Array.isArray(references)) {
+				return references.map(reference => {
+					return <modes.IReferenceInformation>{
```

Comment:
In TypeScript /JavaScript arrow functions work like lambda expressions in functional languages, meaning without curly braces it returns an expression:
```ts
[1, 2, 3].map(x => x * 2) // [2, 4, 6]

fetch('whatever').then(res => res.json())

// note the paranthesis here so the object literal is not treated as a function body
references.map(reference => ({
  foo: reference.bar
}))
```